test/run-tests

```diff
@@ -861,23 +893,89 @@ def main():
         default=False,
         action="store_true",
     )
+    parser.add_argument(
+        "--keep-results",
+        help="For debugging - do not remove the temp results directory",
+        default=False,
+        action="store_true",
+    )
 
     args = parser.parse_args()
 
     # default values for config
     config = {"repositories": [], "images": [], "name": "linux-system-roles-test"}
 
+    # check for settings via env. vars
+    args.secrets = os.environ.get("TEST_HARNESS_SECRETS", args.secrets)
+    args.config = os.environ.get("TEST_HARNESS_CONFIG", args.config)
+    args.cache = os.environ.get("TEST_HARNESS_CACHE", args.cache)
+    args.inventory = os.environ.get("TEST_HARNESS_INVENTORY", args.inventory)
+    args.pull_request = os.environ.get("TEST_HARNESS_PULL_REQUEST", args.pull_request)
+    args.use_images = os.environ.get("TEST_HARNESS_USE_IMAGES", args.use_images)
+    args.dry_run = os.environ.get("TEST_HARNESS_DRY_RUN", args.dry_run)
+    args.keep_results = os.environ.get("TEST_HARNESS_KEEP_RESULTS", args.keep_results)
+
     with open(args.config + "/config.json") as configfile:
         config.update(json.load(configfile))
         images = config["images"]
         random.shuffle(images)
         repos = [r.split("/") for r in config.get("repositories", [])]
         random.shuffle(repos)
 
+    # this is the name for github status reporting
+    if "variant" not in config:
+        if config.get("name") == "linux-system-roles-test-staging":
+            config["variant"] = "staging"
+        else:
+            config["variant"] = ""
+    config["variant"] = os.environ.get("TEST_HARNESS_CONFIG_VARIANT", config["variant"])
+
+    env_repos = os.environ.get("TEST_HARNESS_REPOSITORIES")
+    if env_repos:
+        repos = [r.split("/") for r in env_repos.split(",")]
     setup_logging(config.get("logging", {}))
 
     check_environment()
 
+    ansible_version = LooseVersion(get_ansible_version())
```

Comment:
It might make sense to move this into its own function to be able to easily create some unit tests to assert that the special cases are handled properly.

Reply:
ugh - this is turning into massive feature creep :-(

Reply:
Another idea is to use the old logic but adjust the debug message to say "image ... does not match pattern ..." without stating whether or not it will be used for testing. Then it would be correct, too.

Reply:
The idea of a unit test was to help you not to make it harder for you. :-)

Reply:
Sure - I'm just noting that we are getting a long way from the original intention of this PR . . .

Reply:
ok - the pattern match logic was fixed in https://github.com/linux-system-roles/test-harness/pull/87/commits/42f209256099726c5e8cf7b18bb3ee5fe7d59084 - I will not add unit testing in this PR - it will have to be done in another PR

Reply:
seems to look correct, now

Reply:
squashed and rebased the PR